### PR TITLE
Move cli authentication to config package

### PR DESCRIPTION
**Pull request by viveksyngh**
Signed-off-by: Vivek Singh <vivekkmr45@yahoo.in>

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Fixes: #797 

## How Has This Been Tested?

I have tested this on my mac with local openfaas deployment. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
This will need a rebase, but whilst we are here. Since the auth package only references the proxy package, I would have suggested adding it to proxy instead of to config?

**Comment by viveksyngh** [comment]
I will rebase it and see if we can move this proxy package or not.

**Comment by LucasRoesler** [comment]
Looks good again

File: proxy/auth.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	token string
+}
+
+func (c *BearerToken) Set(req *http.Request) error {
+	req.Header.Set("Authorization", "Bearer "+c.token)
+	return nil
+}
+
+//NewCLIAuth returns a new CLI Auth
+func NewCLIAuth(token string, gateway string) ClientAuth {
```
Comment (alexellis) [inline proxy/auth.go:38]: Can we change to `(ClientAuth, error)`?

**Comment by alexellis** [comment]
Thank you for making the changes. Will there be any changes required in OFC or the certifier? (the clients of the SDK)

**Comment by viveksyngh** [comment]
Yes, we will have to update wherever we have used CLI SDK because NewClient function return type is also updated.




Get Outlook for Android







On Fri, May 8, 2020 at 1:22 AM +0530, "Alex Ellis" <notifications@github.com> wrote:















Thank you for making the changes. Will there be any changes required in OFC or the certifier? (the clients of the SDK)

**Comment by viveksyngh** [comment]
I will create an issue and take care of them as well.

**Pull request merged**